opendcs-web-client/src/main/webapp/WEB-INF/web.xml

```diff
@@ -28,8 +28,16 @@
         <welcome-file>default.jsp</welcome-file>
     </welcome-file-list>
     <context-param>
-        <param-name>api_base_url</param-name>
-        <param-value>http://opendcsapi:8080/odcsapi</param-value>
+        <param-name>api_base_path</param-name>
+        <param-value>odcsapi</param-value>
+    </context-param>
+    <context-param>
+        <param-name>authentication_type</param-name>
+        <param-value>basic</param-value>
+    </context-param>
+    <context-param>
+        <param-name>authentication_base_path</param-name>
```

Comment:
This really shouldn't need to be configured, with the changes @adamkorynta just merged in there should be sufficient feedback from the API itself to properly redirect, or list the available choices to login.

Reply:
The REST API doesn't return a login URL (or list of URL's). It doesn't seem right that the REST API would know about the `CWMSLogin/login` endpoint, it just needs to know that the JSESSIONIDSSO was setup with an appropriate principal. If the web client, for example, were to run queries against both OpenDCS REST API and CDA, the CWMSLogin endpoint would be the same authentication for both. Would we want the client to interrogate both API's and if they support the same auth type and login URL, then only make a single call?

Reply:
That's were things get a bit.. obnoxious? for us. The whole cookie thing (JSESSIONIDSSO and some of these context parameter) are really geared towards everything being from the same server.

Unfortunately for the original designers of that specification things like external identity providers (OpenID Connect) became a thing.

I suspect the OpenAPI specification is the reasonable balance. Avoids us having to design a custom JSON and write the endpoints but still provides the requisite information about the various options.

At the same time I'm not super happy with some of the limits I've felt in the spec, at least as to what the swagger will render. The actual Authorization header spec is relatively free form and specifically asserts there should be flexibility but the html/js will only render the defined standards correctly.

Reply:
Sounds good - 

I've referenced this conversation in the issue for swagger annotations:

https://github.com/opendcs/rest_api/issues/78